base/stream.jl

```diff
@@ -1100,7 +1103,7 @@ function uv_write(s::LibuvStream, p::Ptr{UInt8}, n::UInt)
     if status < 0
         throw(_UVError("write", status))
     end
-    return Int(n)
+    return n % Int  # Safe (on 32-bit too, because of check at the top)
```

Comment:
So if `n` was larger than `typemax(Int)` (i.e. more than 2 GB), the old code would throw an exception, the new code instead returns a negative value. Based on your PR description (but not the title), is that perhaps you true goal? To not have the code throw here?

Reply:
Note two cases a) 32-bit and b) 64-bit:

a) No it should not return a negative value because of the check at the top making it impossible.

b) On 64-bit strings of length 2^63 (and larger, i.e. "impossible" lengths) would have returned negative with this change, but Inexact as it was, and pay for that check if it passes or not.